msgs-spray-json/src/main/scala/com/thenewmotion/ocpi/msgs/sprayjson/v2_1/DefaultJsonProtocol.scala

```diff
@@ -127,7 +128,16 @@ trait DefaultJsonProtocol extends spray.json.DefaultJsonProtocol {
     }
   }
 
-  implicit val displayTextFormat = jsonFormat2(DisplayText)
+  implicit val displayTextFormat: RootJsonFormat[DisplayText] = jsonFormat2(DisplayText)
+
+  def successRespReader[D: JsonReader]: RootJsonReader[SuccessResp[D]] = ???
+
+  def successPagedRes[T: JsonFormat]: RootJsonReader[SuccessResp[Iterable[T]]] =
+    js => {
+      val withData = successRespFormat[Iterable[T]].read _
+      val withoutData = successRespUnitFormat.read(_: JsValue).copy(data = Iterable.empty[T])
+      Try(withData(js)).recover { case _ => withoutData(js) }.get
```

Comment:
What happens when there is an error in one of the elements of the data-array? Does it then silently turn data into an empty array, instead of rejecting the json?

Reply:
My understanding was that the `recover { .... }` can throw as well and if it does then whole `Try(...).recover(...)` is a `Failure(...)` which will cause the final `.get` to throw. 
From a REPL session:
```
Try[Int](throw new Exception).recover[Int]{ case _ => 1/0 }.get

java.lang.ArithmeticException: divide by zero
  at $anonfun$1.applyOrElse(<console>:13)
  at $anonfun$1.applyOrElse(<console>:13)
  at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:38)
  at scala.util.Failure.recover(Try.scala:234)
  ... 33 elided
  ```